.travis.yml

```diff
@@ -20,6 +19,27 @@ install:
   - rustup target add riscv32imc-unknown-none-elf
   - rustup component add rustfmt
   - rustup component add clippy
+  - cargo install -f elf2tab --version 0.4.0
+  # Ubuntu doesn't package QEMU for RISC-V, so let's build it
+  - wget https://download.qemu.org/qemu-4.2.0.tar.xz
+  - tar xJf qemu-4.2.0.tar.xz
+  - pushd qemu-4.2.0
+  - ./configure --target-list=riscv32-softmmu
+  - make -j8
+  - popd
+  # Build Tock, it needs to be outside of the libtock-rs source
+  - pushd ../
+  - git clone --depth=50 https://github.com/tock/tock.git
+  - cd tock/boards/hifive1
+  # Use a known working version of Tock
+  - git checkout c94059d3e25dc635e682facff4894ef43b9aca0e
+  - make
+  - popd
 
 script:
   - ./run_all_checks.sh
+  # Run a QEMU instance of the HiFive1 app
+  - PLATFORM=riscv32 cargo build --target=riscv32imac-unknown-none-elf --example hello_world
+  - elf2tab -o ./Tock-app.tab ./target/riscv32imac-unknown-none-elf/debug/examples/hello_world --stack 2048 --app-heap 1024 --kernel-heap 1024 -v --protected-region-size=64
+  - timeout --foreground 10s ./qemu-4.2.0/riscv32-softmmu/qemu-system-riscv32 -M sifive_e -kernel ../tock/boards/hifive1/target/riscv32imac-unknown-none-elf/release/hifive1 -device loader,file=./target/riscv32imac-unknown-none-elf/debug/examples/hello_world.tbf,addr=0x20430000 -nographic | tee serial
```

Comment:
This script was intended to be run locally. I would leave the integration tests out for now as they involve compiling qemu.

Reply:
I'm not clear what you mean here

Reply:
Just a mistake, I thought these lines were in the run all checks script. Sorry.